### PR TITLE
Added CMakeLists.txt to build and install with cmake

### PR DESCRIPTION
**Pull request by ferkulat**

### CODE REVIEW

File: CMakeLists.txt
Comment (pinotree) [inline CMakeLists.txt]: You do not need to create a `check` target manually, ctest already creates a `test` one.

File: CMakeLists.txt
Comment (pinotree) [inline CMakeLists.txt]: This will install it to `${CMAKE_INSTALL_PREFIX}/man`, so e.g. `/usr/man` when setting `/usr` as prefix.  Please use `share/man` instead, or even better make use of the `GNUInstallDirs` cmake module (available in cmake >= 2.8.5) to get the right installation directories on standard systems (and that can be overridden by users).

File: CMakeLists.txt
Comment (pinotree) [inline CMakeLists.txt]: Most probably the required cmake version can be lowered a lot, since this `CMakeLists.txt` is very simple, and it does not use many advanced features. IMHO 3.0 should be already good enough, as I do not remember which version introduced the `LINK_PUBLIC` feature for `target_link_libraries`.

**Review by pinotree — CHANGES_REQUESTED** [review]
I am not the libcsv maintainer, but IMHO there are few things that ought to be changed/improved before merging.

**Comment by ferkulat** [comment]
Hi @rgamble  are there still any objections about this pull request?

**Pull request closed without merging**
